src/editor/Store/FileExplorerStoreConnector.re

```diff
@@ -0,0 +1,34 @@
+/**
+   FileExplorerStoreConnector.re
+
+   Implements an updater (reducer + side effects) for the File Explorer
+ */
+open Oni_Model;
+
+let start = () => {
+  let (stream, _) = Isolinear.Stream.create();
+
+  let getExplorerFilesEffect = (cwd, languageInfo, iconTheme, ignored) => {
+    Isolinear.Effect.createWithDispatch(~name="explorer.open", dispatch => {
+      let newTree =
+        FileExplorer.getDirectoryTree(cwd, languageInfo, iconTheme, ignored);
+      dispatch(Actions.SetExplorerTree(newTree));
+    });
+  };
+
+  let updater = (state: State.t, action: Actions.t) => {
+    switch (action) {
+    | OpenExplorer(directory) => (
+        state,
+        getExplorerFilesEffect(
+          directory,
+          state.languageInfo,
+          state.iconTheme,
+          ["node_modules", "_esy"],
```

Comment:
Should we make an issue to add in a config option for this?

Ie, hooking up `"files.exclude"` like in VSCode.

Its a decent good first issue as well I think.

Reply:
definitely :+1:, I just got lazy at that point/didn't want to touch anymore files so this PR didn't get too big (which is a bad excuse, was primarily laziness)

Reply:
decided to fix this here rather than create more work later

Reply:
@CrossR seems like it might be better to read `.gitignore` in this case?

Reply:
Could be worth opening an issue for a follow up for that!

Adding it now doesn't make much sense since I think it would be a bigger task, since we'd need to either load/parse a users' gitignore, or use a tool that does that for us. We get it for free with the QuickOpen, but not here unfortunately.

(Its free for the QuickOpen as that uses ripgrep that automatically uses `.gitignore` files)

Reply:
yeah, doing `.gitignore` stuff is definitely out of scope for the original implementation

Reply:
Hmm. then... I think specialcasing those two isn't a great idea (seems like you're still specialcasing them in `default`), plus it might be a good performance test

Reply:
VSCode does come with a default list:

```json
"files.exclude": {
  "**/.git": true,
  "**/.svn": true,
  "**/.hg": true,
  "**/CVS": true,
  "**/.DS_Store": true
}
```

`node_modules` is usually excluded as well, and since this is a reason project `_esy` makes sense too.

We can always add a performance benchmark as part of the follow up tasks. Would probably fit well with the checks around listing the home folder etc (ie making sure we bail out if we start recursing too far).

This all seems to make sense for the base implementation though!